### ISSUE TITLE
Help could be more explanitory regarding 'new' subcommand

### ISSUE THREAD

**Issue by Phrohdoh** [rust-lang/cargo#3473]
# Issue:

ref: https://botbot.me/mozilla/rust/2016-12-29/?msg=78592153&page=20

tldr:
`cargo new --bin --name my_project` is invalid.
The help covers why but could be more explicit.

# Info:

What this person actually wanted was `cargo new --bin my_project` which creates the `my_project` directory with a project named `my_project` inside of it.

The full version of that would be `cargo new --bin --name my_project my_project`.

Help for `new` should note that if `--name` is not given the package name will default to the name of the path given.

I feel there should be an example illustrating the use of `--name` which would probably prevent this in the future.

Example:

```
$ cargo bin --new ripag rust_projects/text_searching/ripag
# This will create rust_projects/text_searching/ripag/[.. project files here .. ]
```

# Resolution:
Originally I had slapped together this change:

`    --name NAME         Set the resulting package name` -> `    --name NAME         Set the resulting package name (defaults to the value of <path>)`

But none of the other flag descriptions mention default values so this felt out of place.
What would be the best way to improve on this situation?
